File: loop_transformations/sources/fuse.1.c
/*
* @@name:	fuse.1
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
#include <math.h>

void func(int n, double* sines, double* cosines)
{
  #pragma omp fuse
  {
    for (int i = 0; i < n; ++i)
      sines[i] = sin(2.0*M_PI*i/n);
    for (int j = 0; j < n; ++j)
      cosines[j] = cos(2.0*M_PI*j/n);
  }
}

void func_equivalent(int n, double* sines, double* cosines)
{
  for (int ij = 0; ij < n; ++ij) {
    sines[ij]   = sin(2.0*M_PI*ij/n);
    cosines[ij] = cos(2.0*M_PI*ij/n);
  }
}

File: loop_transformations/sources/fuse.1.f90
! @@name:       fuse.1
! @@type:       F-free
! @@operation:  compile
! @@expect:     success
! @@version:    omp_6.0

subroutine func(n, sines, cosines)
  implicit none
  integer,          intent(in)  :: n
  double precision, intent(out) :: sines(n), cosines(n)
  double precision, parameter   :: M_PI=4.0d0*DATAN(1.0d0)
  integer :: i, j

  !$omp fuse
    do i = 1, n
      sines(i) = sin(2.0d0*M_PI*(i-1)/n)
    end do
    do j = 1, n
      cosines(j) = cos(2.0d0*M_PI*(j-1)/n)
    end do
  !$omp end fuse
end subroutine

subroutine func_equivalent(n, sines, cosines)
  implicit none
  integer,          intent(in)  :: n
  double precision, intent(out) :: sines(n), cosines(n)
  double precision, parameter   :: M_PI=4.0d0*DATAN(1.0d0)
  integer :: ij
    do ij = 1, n
      sines(ij)   = sin(2.0d0*M_PI*(ij-1)/n)
      cosines(ij) = cos(2.0d0*M_PI*(ij-1)/n)
    end do
end subroutine
